src/System.Management.Automation/help/CommandHelpProvider.cs

```diff
@@ -542,6 +548,13 @@ private string GetHelpFile(string helpFile, CmdletInfo cmdletInfo)
         /// <returns></returns>
         private string FindHelpFile(CmdletInfo cmdletInfo)
         {
+            if (InternalTestHooks.BypassOnlineHelpRetrieval)
+            {
+                // By returning null, we force get-help to generate a metadata driven help object,
+                // which includes a helpUri that points to the fwlink defined in the cmdlet code.
```

Comment:
> 1) get-help will not find a help file; instead, it will generate a metadata driven object.
> 2) get-help -online <cmdletName>  will return the helpuri instead of opening the default web browser.

Yes, I know this test hook is used in that test, but it's used there for `get-help -online` to return helpuri instead of opening web browser. I don't see where it's used for getting metadata driven help object. Can you please clarify?